src/ArduinoIoTCloudTCP.cpp

```diff
@@ -581,6 +580,30 @@ void ArduinoIoTCloudTCP::sendPropertiesToCloud()
     }
 }
 
+void ArduinoIoTCloudTCP::sendPropertiesToCloud()
+{
+  sendPropertyContainerToCloud(_property_container);
+}
+
+void ArduinoIoTCloudTCP::sendOTAPropertiesToCloud()
+{
+  PropertyContainer to_property_container;
+  PropertyContainer from_property_container = _property_container;
```

Comment:
Also: Is it really necessary to copy the full property container here? Since you do not change it I don't see why you shouldn't be able to use `_property_container` in the `std::for_each` loop below.

Reply:
That was also my intention, but doing so compiler is complaining...
```
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp: In member function 'void ArduinoIoTCloudTCP::sendOTAPropertiesToCloud()':
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp:596:19: error: capture of non-variable 'ArduinoIoTCloudClass::_property_container' 
                 [&_property_container, &to_property_container ] (String const & name)
                   ^~~~~~~~~~~~~~~~~~~
In file included from /home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.h:27:0,
                 from /home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp:25:
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloud.h:147:23: note: 'PropertyContainer ArduinoIoTCloudClass::_property_container' declared here
     PropertyContainer _property_container;
                       ^~~~~~~~~~~~~~~~~~~
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp: In lambda function:
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp:598:45: error: 'this' was not captured for this lambda function
                   Property* p = getProperty(_property_container, name);
                                             ^~~~~~~~~~~~~~~~~~~
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp:598:45: error: invalid use of non-static data member 'ArduinoIoTCloudClass::_property_container'
In file included from /home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.h:27:0,
                 from /home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloudTCP.cpp:25:
/home/pennam/Arduino/libraries/ArduinoIoTCloud/src/ArduinoIoTCloud.h:147:23: note: declared here
     PropertyContainer _property_container;
                       ^~~~~~~~~~~~~~~~~~~
```
I think the point is that `_property_container` is not a local variable, but this was actually my firs lambda expression so i'm not really sure. :sweat_smile:

Reply:
That one is easy :+1: you need to capture `this` within the lamda :wink: 
```C++
[this, &to_property_container ] (String const & name)
```

Reply:
nice, I could have spent weeks trying to find this solution. Thanks Alex.  :rocket: 

Now i'm ready to fix it, merge everything and release a new library version.